### DISCUSSION TITLE
Feature Request: Support ORC Files Natively

### DISCUSSION THREAD

**Discussion by voycey** [duckdb/duckdb#5979] [Ideas]
Many of our files are in ORC format as we use TrinoDB in our workflow, we have found that ORC performs better due to predicate pushdown support and more efficient encryption on large datasets. we are investigating DuckDB as an alternative to a traditional cluster type setup and would be great to be able to directly use our files rather than re-convert everything.

Not sure how re-usable it would be but the team at Starburst and Trino have invested significant time into the ORC Format over the years (https://trino.io/blog/2019/04/23/even-faster-orc.html) and I have found that (at least in the tooling we used) ORC provided a significant performance and cost benefit over Parquet (this was however several years back that I last tested it so I am sure things have improved since).

I assume this is already supported in some format via arrow (https://duckdb.org/2021/12/03/duck-arrow.html) as arrow has an ORC reader, I am not sure how efficient this vs the Trino ORC library which has had heavy investment over the years.

Thoughts and discussion welcome :)

**Comment by hannes** [comment]
Hey we have not had many requests, but would certainly consider adding the functionality. Feel free to reach out to hannes@duckdblabs.com to discuss options. Cheers

**Comment by aryanRdy** [comment]
Yeah, I have the same requirement, We have same setup as @voycey. It would be great to have ORC support natively
